lib/srv/db/server.go

```diff
@@ -646,6 +659,15 @@ func (s *Server) handleConnection(ctx context.Context, conn net.Conn) error {
 	if err != nil {
 		return trace.Wrap(err)
 	}
+
+	clientIP := sessionCtx.Identity.ClientIP
```

Comment:
What happens if this is empty? Which will be the case if an older proxy (that doesn't set ClientIP in the identity) connects to this database agent.

Reply:
@r0mant Good point. I updated the logic. Now if the client doesn't send the IP rate limiting will be disabled (to not change the current behavior) and a debug log will be produced (I'm not sure if debug is the best in this case. Logging a warning for every connection sounds a bit excessive).

Reply:
@jakule Let's add a TODO comment here saying that we should start requiring client IP to be set starting Teleport 10 and this is only for backwards compatibility.